### PR TITLE
Offset workaround, New Class Hierarchy (Catroid Develop) 

### PR DESCRIPTION
**Pull request by cfkh**
Offset workaround was tested with several Projects. Temporarily looking good.
For creation of new class hierarchy no new rules had to be added. Also no new manual stuff.
Everything was tested with the test suite.

### CODE REVIEW

File: src/scratchtocatrobat/tools/svgtopng.py [hunk 1]
```diff
@@ -324,13 +340,14 @@ def _parse_and_rewrite_svg_file(svg_input_path, svg_output_path):
...
-
+    
+    check = False
     while True:
         read_line = buffered_reader.readLine()
 
         if read_line is None:
             break
-
+
```
Comment (rsamer) [inline src/scratchtocatrobat/tools/svgtopng.py]: Please redundant indentation.

File: src/scratchtocatrobat/tools/svgtopng.py [hunk 2]
```diff
@@ -341,16 +358,23 @@ def _parse_and_rewrite_svg_file(svg_input_path, svg_output_path):
             if view_box_values[1] != 0:
                 view_box_values[3] += view_box_values[1]
                 view_box_values[1] = 0
-
-            new_view_box = str(view_box_values[0]) + " " + str(view_box_values[1]) + " " + \
-                           str(view_box_values[2]) + " " + str(view_box_values[3])
-            read_line = re.sub(r"viewBox=\"[\-|0-9| ]+\"", "viewBox=\""
-                               + new_view_box + "\"", read_line, 1)
+
```
Comment (rsamer) [inline src/scratchtocatrobat/tools/svgtopng.py]: same here

File: src/scratchtocatrobat/tools/svgtopng.py [hunk 3]
```diff
@@ -341,16 +358,23 @@ def _parse_and_rewrite_svg_file(svg_input_path, svg_output_path):
...
                 view_box_values[3] += view_box_values[1]
                 view_box_values[1] = 0
-
-            new_view_box = str(view_box_values[0]) + " " + str(view_box_values[1]) + " " + \
-                           str(view_box_values[2]) + " " + str(view_box_values[3])
-            read_line = re.sub(r"viewBox=\"[\-|0-9| ]+\"", "viewBox=\""
-                               + new_view_box + "\"", read_line, 1)
+            
+            #new_view_box = str(view_box_values[0]) + " " + str(view_box_values[1]) + " " + \
+            #               str(view_box_values[2]) + " " + str(view_box_values[3])
```
Comment (rsamer) [inline src/scratchtocatrobat/tools/svgtopng.py]: Please get rid of comments containing stale code fragments/snippets

File: src/scratchtocatrobat/tools/svgtopng.py [hunk 4]
```diff
@@ -37,6 +37,18 @@
...
+from org.w3c.dom import Element
+from org.apache.batik.util import XMLResourceDescriptor
+from org.apache.batik.dom.svg import SAXSVGDocumentFactory
+from org.apache.batik.dom.svg import SVGDOMImplementation
+from org.apache.batik.dom.svg import SVGOMDocument
+import java.io.InputStream
+import java.io.File
+import java.nio.file.Files
+'''
+
```
Comment (rsamer) [inline src/scratchtocatrobat/tools/svgtopng.py]: Please remove unused imports

**Comment by cfkh** [comment]
Should be updated according to comments.

**Pull request merged**